### PR TITLE
Initial project

### PR DESCRIPTION
**Pull request by austinkelleher**
This is not ready for a very formal review. There are no tests and things aren't completely working. I'm opening this PR as more of a way to share my current directory structure. I propose that we follow something similar to this.

### CODE REVIEW

File: src/config/config.development.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+httpServerPort: 3000
+amqUrl: 'amqp://127.0.0.1:5672'
+redisClusterNodes:
+  - host: '127.0.0.1'
+    port: 7000
+  - host: '127.0.0.1'
+    port: 7000
```
Comment (ctdio) [inline src/config/config.development.yml]: 7001

File: package.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    "test": "npm run docker:build && docker-compose run test && docker-compose down",
+    "test:unit": "nyc ava test/unit",
+    "test:integration": "docker-compose run test npm run test:docker:integration",
+    "test:docker:ci": "npm run lint && nyc ava test/unit test/integration --verbose",
+    "test:docker:integration": "nyc ava test/unit test/integration --verbose",
+    "start": "browser-refresh src/server.js",
+    "cluster:start": "browser-refresh src/cluster.js"
+  },
+  "author": "",
+  "license": "ISC",
```
Comment (ctdio) [inline package.json]: No MIT?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+version: '3'
+services:
+  postgres:
+    image: 'postgres:9.6.3-alpine'
+    environment:
+      POSTGRES_USER: 'postgres'
+      POSTGRES_PASSWORD: 'postgres'
+      POSTGRES_DB: 'windbreaker'
+    ports:
+      - '5432:5432'
```
Comment (ctdio) [inline docker-compose.yml]: You may encounter errors when running CI with exposed ports like this.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    "docker:build": "docker-compose build",
+    "docker:start": "docker-compose up rediscluster rabbitmq",
+    "docker:stop": "docker-compose down",
+    "lint": "eslint .",
+    "test": "npm run docker:build && docker-compose run test && docker-compose down",
+    "test:unit": "nyc ava test/unit",
+    "test:integration": "docker-compose run test npm run test:docker:integration",
+    "test:docker:ci": "npm run lint && nyc ava test/unit test/integration --verbose",
+    "test:docker:integration": "nyc ava test/unit test/integration --verbose",
+    "start": "browser-refresh src/server.js",
```
Comment (ctdio) [inline package.json]: Due to a limitation of the docker image we use for rediscluster, you won't be able to actually execute commands on the cluster from outside of docker.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "windbreaker-server",
```
Comment (ctdio) [inline package.json]: `windbreaker-webhooks`

File: src/producer/index.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  logger.info('Sucessfully started webhook event producer!')
+
+  producer.on('error', (err) => {
+    logger.error(`Producer error: `, err)
+
+    // TODO: Write test to ensure that this bubbles up to an uncaught exception
+
+    startProducer()
+      // .then(() => {
+      // TODO: Handle all messages in async queue here
```
Comment (ctdio) [inline src/producer/index.js]: Don't forget to reattach an error listener to the new producer that gets created.

File: src/server.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+})()
+
+process.on('uncaughtException', function (err) {
+  // TODO: Log this to a database
+  console.error('Uncaught exception: ', err)
+  process.exit(1)
+})
+
+process.on('unhandledException', function (err) {
+  // TODO: Log this to a database
```
Comment (ctdio) [inline src/server.js]: For situations like this, there's not real guarantee that we will have access to a our database at this point. Using a monitor or logging system like newrelic or scaylr would be slightly more reliable.

File: package.json [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    "test:integration": "docker-compose run test npm run test:docker:integration",
+    "test:docker:ci": "npm run lint && nyc ava test/unit test/integration --verbose",
+    "test:docker:integration": "nyc ava test/unit test/integration --verbose",
+    "start": "browser-refresh src/server.js",
+    "cluster:start": "browser-refresh src/cluster.js"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "custom-logger": "^0.3.1",
```
Comment (ctdio) [inline package.json]: remove

File: package.json (hunk 4 above)
Comment (Druotic) [reply package.json]: It's actually a limitation of redis cluster in NAT'd environments. Redis cluster will advertise IPs/ports which are routable from inside the docker network, but the host machine has no idea how to route to those addresses. The initial connect will work, but any requests which try to look up a key that is on a different master will result in the client getting redirected to an address which will be unroutable from the perspective of the host. Related: https://github.com/antirez/redis/issues/2527

**Review by Druotic — COMMENTED** [review]
👍

**Review by Druotic — APPROVED** [review]

**Pull request merged**